### ISSUE TITLE
bug: Dockerfile.ci: broken clang-15 installation in v0.26.0

### ISSUE THREAD

**Issue by avolkov-1221** [zephyrproject-rtos/docker-image#144]
Hello,

After migration to Ubuntu 22.04, the vanilla clang installation been broken due to a conflict with Ubuntu's llvm-15:i386 (installed as a dependency of 'libsdl2-dev:i386'). Actually, the bug is masked by a second "false positive installation" error: an incorrect version of the repository being used for downloading causes vanilla clang from 'apt.llvm.org' to be completely ignored.

The second bug is in line 55:
`echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy main" | tee /etc/apt/sources.list.d/llvm-official.list && \`

while it should be (according to https://apt.llvm.org/):
`echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-${LLVM_VERSION} main" | tee /etc/apt/sources.list.d/llvm-official.list && \`

With the above modification, the correct clang version `15.0.7~++20230131104537+8dfdcc7b7bf6-1~exp1~20230131104626.110` started to install and the Docker build process immediately failed due to version's conflict.

**Comment by KloudJack** [comment]
Good catch! Thanks for doing the legwork on tracking down the source of this issue.

The second bug explains the difference I'd experienced trying to install LLVM via the `llvm.sh` script, as the script sets up the correct LLVM 15 Jammy APT repository source, and would lead to the same install failure you've described.

1. Since the official `jammy-updates` repository provides LLVM 15, I wonder if we should just ship that version. IMHO this would simplify the Dockerfiles and decrease future growing pains when moving to newer Ubuntu releases.

2. We could also potentially drop the `libsdl2-dev:i386` dependency, I've tracked it back to commit 6ca77308a4ed36c3d5d013d265e17753c36b7ccc (from 2018)
    Unfortunately there's no mention on the purpose behind adding the package, though it seems to have been added in tandem with VNC support for displaying `native-posix` target framebuffer applications.
    Could @nashif or @vanwinkeljan give feedback on this?

I'll test the necessity of `libsdl2-dev:i386` for `samples/subsys/display/cfb` on `native_posix` and put together a PR with either one or both of the above changes based on the outcome.

**Comment by avolkov-1221** [comment]
> Good catch! Thanks for doing the legwork on tracking down the source of this issue.
> 
> The second bug explains the difference I'd experienced trying to install LLVM via the `llvm.sh` script, as the script sets up the correct LLVM 15 Jammy APT repository source, and would lead to the same install failure you've described.
> 
>     1. Since the official `jammy-updates` repository provides LLVM 15, I wonder if we should just ship that version. IMHO this would simplify the Dockerfiles and decrease future growing pains when moving to newer Ubuntu releases.

Agree, actually I already use it. Or we could move to the latest, 16/17 vanilla version, but I'm afraid that would require massive re-testing of everything, with unclear reasons and unpredictable results.

> 
> 
> ~2. We could also potentially drop the `libsdl2-dev:i386` dependency, I've tracked it back to commit [6ca7730](https://github.com/zephyrproject-rtos/docker-image/commit/6ca77308a4ed36c3d5d013d265e17753c36b7ccc) (from 2018) Unfortunately there's no mention on the purpose behind adding the package, though it seems to have been added in tandem with VNC support for displaying `native-posix` target framebuffer applications. Could @nashif or @vanwinkeljan give feedback on this?~ `libsdl2-dev:i386` is required for cfb on `native_posix`.
> 
Damn, ok. let it keep to live.

> I'll put together a PR with the above change.

**Comment by stephanosio** [comment]
Since Clang 16 has been released and there is no reason why we must keep using Clang 15, I will fix this by updating the Dockerfile to install Clang 16 using the official `llvm.sh` script.
